HKDF based on SHA3-512 fails to build

### Zig Version

0.13.0

### Steps to Reproduce and Observed Behavior

When trying to create an HKDF based on SHA3-512, code fails to build due to an assertion `assert(out.len <= prk_length * 255); // output size is too large for the Hkdf construction` that ends up overflowing an `u11` from `std.crypto.hash.sha3.Keccak`.

To reproduce, any code calling `expand` on such an HKDF instance will fail to build:

```
$ cat <<EOF > hkdf-issue.zig
const std = @import("std");
test "hkdf" {
    const sha3_512 = std.crypto.hash.sha3.Sha3_512;
    const hmac_sha3_512 = std.crypto.auth.hmac.Hmac(sha3_512);
    const hkdf = std.crypto.kdf.hkdf.Hkdf(hmac_sha3_512);
    const prk = hkdf.extract("", "");
    var out = [1]u8{0};
    hkdf.expand(out[0..], "", prk);
}
EOF

$ zig test hkdf-issue.zig
zig/zig-linux-aarch64-0.13.0/lib/std/crypto/hkdf.zig:42:42: error: overflow of integer type 'u11' with value '16320'
            assert(out.len <= prk_length * 255); // output size is too large for the Hkdf construction
                              ~~~~~~~~~~~^~~~~
referenced by:
    test.hkdf: hkdf-issue.zig:8:9
    remaining reference traces hidden; use '-freference-trace' to see all reference traces
```

Changing the failed assertion to `assert(out.len <= @as(usize, prk_length) * 255);` seems to fix the issue.

### Expected Behavior

Expected behavior: I should be able to construct an HKDF with SHA3-512 and not encounter a build issue.

--- a/lib/std/crypto/hkdf.zig
+++ b/lib/std/crypto/hkdf.zig
@@ -88,3 +88,12 @@ test "Hkdf" {
     hkdf.final(&prk2);
     try htest.assertEqual("077709362c2e32df0ddc3f0dc47bba6390b6c73bb50f9c3122ec844ad7c2b3e5", &prk2);
 }
+
+test "Hkdf Sha3-512" {
+    const sha3_512 = std.crypto.hash.sha3.Sha3_512;
+    const hmac_sha3_512 = hmac.Hmac(sha3_512);
+    const hkdf = Hkdf(hmac_sha3_512);
+    const prk = hkdf.extract("", "");
+    var out = [1]u8{0};
+    hkdf.expand(out[0..], "", prk);
+}

--- a/lib/std/crypto/sha3.zig
+++ b/lib/std/crypto/sha3.zig
@@ -52,7 +52,7 @@ pub fn Keccak(comptime f: u11, comptime output_bits: u11, comptime default_delim
         st: State,
 
         /// The output length, in bytes.
-        pub const digest_length = output_bits / 8;
+        pub const digest_length = std.math.divCeil(comptime_int, output_bits, 8) catch unreachable;
         /// The block length, or rate, in bytes.
         pub const block_length = State.rate;
         /// The delimiter can be overwritten in the options.

